### PR TITLE
Add Helm chart for Slack plugin

### PR DESCRIPTION
**Pull request by marshall-lee**
Related to #356 and #270

Requires #319

### CODE REVIEW

**Review by Joerger — APPROVED** [review]
I'm not too familiar with Helm Chart architecture, but it LGTM.

**Comment by tcsc** [comment]
@webvictim - Would you mind having a peek at this?

File: access/slack/charts/teleport-access-slack/values.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+##################################################
+# Values that must always be provided by the user.
+##################################################
+
+# The name of Kubernetes service of Teleport.
```
Comment (webvictim) [inline access/slack/charts/teleport-access-slack/values.yaml:5]:
```suggestion
# The name of the Teleport service in the Kubernetes cluster.
```

File: access/slack/charts/teleport-access-slack/values.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+# Container image for the plugin.
+image: quay.io/gravitational/teleport-access-slack
+# Log level for the plugin process.
+# Available log levels are: DEBUG, INFO, WARNING, ERROR.
+# The default is INFO, which is recommended in production.
+# DEBUG is useful during first-time setup or to see more detailed logs for debugging.
+logLevel: INFO
+# The port open on Kubernetes service of Teleport.
+# The simplest is to use reverse tunnel port which is 3024.
```
Comment (webvictim) [inline access/slack/charts/teleport-access-slack/values.yaml:28]:
```suggestion
# The open Teleport reverse tunnel port in the Kubernetes cluster. Default: 3024
```

File: access/slack/charts/teleport-access-slack/values.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+##################################
+
+# Affinity for pod assignment
+# https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#affinity-and-anti-affinity
+affinity: {}
+
+# Kubernetes annotations to apply
+# https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/
+annotations:
+  # Annotations for the ConfigMap (note: these will not be applied in 'custom' mode)
```
Comment (webvictim) [inline access/slack/charts/teleport-access-slack/values.yaml:42]:
```suggestion
  # Annotations for the ConfigMap
```

File: access/slack/charts/teleport-access-slack/templates/config.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    identity = "/var/lib/plugin-identity/identity"
+
+    [slack]
+    token = "/var/lib/plugin-token/token"
+
+    [log]
+    severity = '{{ .Values.logLevel }}'
+    output = "stderr"
+
+
```
Comment (webvictim) [inline access/slack/charts/teleport-access-slack/templates/config.yaml:22]: Remove blank line

```suggestion
```

**Review by webvictim — APPROVED** [review]
LGTM I think.

It would be nice to have a Drone pipeline that automatically runs `helm lint` as part of CI, similar to https://github.com/gravitational/teleport/blob/master/Makefile#L556-L584 - but I don't think we should block this PR for that.

**Comment by marshall-lee** [comment]
@webvictim @tcsc this Helm chart works but it uses custom resources from #319 so it doesn't make sense until #319 is merged.

**Review by tcsc — APPROVED** [review]

**Comment by wadells** [comment]
This has been superseded by https://github.com/gravitational/teleport-plugins/pull/494

**Pull request closed without merging**